### PR TITLE
Add questions for n17681

### PR DESCRIPTION
**Pull request by ryanofsky**

### CODE REVIEW

**Comment by ryanofsky** [comment]
@jnewbery , here are the questions I came up with. Feel free to change or drop anything here that isn't good

File: _posts/2020-05-13-#17681.md [hunk 1]
```diff
@@ -34,7 +34,63 @@ commit:
...
+   version](https://github.com/ryanofsky/bitcoin/commits/review.17681.1) of
+   this PR tried (the implementation was buggy) to store
+   [`m_inactive_hd_chains`](https://github.com/ryanofsky/bitcoin/blob/review.17681.1/src/wallet/scriptpubkeyman.h#L301)
+   map entries as
+   [`"inactivehdchain"`](https://github.com/ryanofsky/bitcoin/blob/review.17681.1/src/wallet/walletdb.cpp#L32)
+   rows in the wallet database. Was it good to stop doing this? What are advantages of
+   storing this information? What are the advantages of not storing it?
+
+10. A [previous
+   version](https://github.com/ryanofsky/bitcoin/commits/review.17681.4) or
```
Comment (jonatack) [inline _posts/2020-05-13-#17681.md:88]: s/or/of/

File: _posts/2020-05-13-#17681.md [hunk 2]
```diff
@@ -34,7 +34,63 @@ commit:
...
+   this PR had a subtle bug on [this
+   line](https://github.com/ryanofsky/bitcoin/blob/review.17681.4/src/wallet/walletdb.cpp#L446).
+   What was the bug and what were the effects?
+
+11. Is test coverage in this PR good? Are there ways the test could be extended?
```
Comment (jonatack) [inline _posts/2020-05-13-#17681.md:93]: For the first sentence, maybe: "Do you think this PR has sufficient test coverage?"

File: _posts/2020-05-13-#17681.md [hunk 3]
```diff
@@ -34,7 +34,63 @@ commit:
...
+1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or
+   NACK?](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)
+   (Don't forget to put your PR review on GitHub.)
+
+2. How does it help users to top up keypools for inactive HD seeds? What is not
+   good about the behavior before this PR?
+
+3. The original motivation for the PR was to remove the restriction that
+   setting a new HD seed wasn't [allowed during Initial Block
+   Download](https://github.com/bitcoin/bitcoin/commit/769b03a83c2aa2b97f344b58dc689be26c6e08e5)?
```
Comment (jonatack) [inline _posts/2020-05-13-#17681.md:46]: It reads like this sentence was intended to be a sentence (the "?" appears to be inadvertent).

File: _posts/2020-05-13-#17681.md [hunk 4]
```diff
@@ -34,7 +34,63 @@ commit:
...
+   as well as existing wallets?
+
+5. The main data structures used in this PR are
+   [`CHDChain`](https://github.com/ryanofsky/bitcoin/blob/review.17681.5/src/wallet/walletdb.h#L88),
+   [`CKeyMetadata`](https://github.com/ryanofsky/bitcoin/blob/review.17681.5/src/wallet/walletdb.h#L126),
+   and
+   [`KeyOriginInfo`](https://github.com/ryanofsky/bitcoin/blob/review.17681.5/src/script/keyorigin.h#L11).
+   What are the purposes of the different data structures?
+
+6. After the wallet has seen a existing key being used, how does the wallet
```
Comment (jonatack) [inline _posts/2020-05-13-#17681.md:62]: s/a/an/

File: _posts/2020-05-13-#17681.md [hunk 5]
```diff
@@ -34,7 +34,63 @@ commit:
...
+
+8. How is the PR structured and divided up? Would it make sense to split up or combine commits?
+
+9. An [earlier
+   version](https://github.com/ryanofsky/bitcoin/commits/review.17681.1) of
+   this PR tried (the implementation was buggy) to store
+   [`m_inactive_hd_chains`](https://github.com/ryanofsky/bitcoin/blob/review.17681.1/src/wallet/scriptpubkeyman.h#L301)
+   map entries as
+   [`"inactivehdchain"`](https://github.com/ryanofsky/bitcoin/blob/review.17681.1/src/wallet/walletdb.cpp#L32)
+   rows in the wallet database. Was it good to stop doing this? What are advantages of
```
Comment (jonatack) [inline _posts/2020-05-13-#17681.md:84]: s/are/are the/

**Comment by jonatack** [comment]
(@ryanofsky, sorry for not seeing this PR yesterday. Let me know if you'd like @jnewbery or I to just go ahead with minor fixup changes and merge this.)

**Comment by jonatack** [comment]
One fixup at the start of the first sentence of the Notes: s/the/The/

**Comment by ryanofsky** [comment]
Thanks @jonatack, it'd be easiest for me if you just merged and made desired edits Just skimmed but all your suggestions seem fine. I've never deleted tags in ryanofsky/bitcoin before, and I'm not planning on deleting these, but agree it'd be better to push tags to the review club repo instead

**Comment by jonatack** [comment]
Ok, it's late here so I'll publish as-is, tweet the announcement, and do a follow-up for the nits. Thanks! Will dive into this PR tomorrow.

**Comment by jonatack** [comment]
Follow-up pushed. Making the announcement. Goodnight!

**Comment by jnewbery** [comment]
Thanks for the questions @ryanofsky and for merging @jonatack

**Pull request merged**